labelCloud/io/labels/kitti.py

```diff
@@ -20,72 +61,142 @@ def __init__(
+                dimensions = tuple([float(v) for v in meta["dimensions"].split()])
                 if self.transformed:
+                    try:
+                        self._get_transforms(pcd_path)
+                    except CalibrationFileNotFound as exc:
+                        logging.exception("Calibration file not found")
+                        logging.warning(
+                            "Skipping loading of labels for this point cloud"
+                        )
+                        return []
+
+                    xyz1 = np.insert(np.asarray(centroid), 3, values=[1])
+                    xyz1 = self.T_c2v @ xyz1
+                    centroid = tuple([float(n) for n in xyz1[:-1]])
                     dimensions = dimensions[2], dimensions[1], dimensions[0]
+                    centroid = (
+                        centroid[0],
+                        centroid[1],
+                        centroid[2] + dimensions[2] / 2,
+                    )  # centroid in KITTI located on bottom face of bbox
                 bbox = BBox(*centroid, *dimensions)
-                if self.transformed:
-                    bbox.set_rotations(
-                        0, 0, rel2abs_rotation(-float(line_elements[14]) + math.pi / 2)
-                    )
-                else:
-                    bbox.set_rotations(0, 0, rel2abs_rotation(float(line_elements[14])))
-                bbox.set_classname(line_elements[0])
-                labels.append(bbox)
+                self.bboxes_meta[id(bbox)] = meta
+                rotation = (
+                    -float(meta["rotation_y"]) + math.pi / 2
+                    if self.transformed
+                    else float(meta["rotation_y"])
+                )
+                bbox.set_rotations(0, 0, rel2abs_rotation(rotation))
+                bbox.set_classname(meta["type"])
+                bboxes.append(bbox)
             logging.info("Imported %s labels from %s." % (len(label_lines), label_path))
-        return labels
+        return bboxes
 
     def export_labels(self, bboxes: List[BBox], pcd_path: Path) -> None:
         data = str()
 
         # Labels
-        for bbox in bboxes:
+        for i, bbox in enumerate(bboxes):
             obj_type = bbox.get_classname()
             centroid = bbox.get_center()
-            if self.transformed:
-                centroid = (-centroid[1], centroid[2] + 2.3, centroid[0])
-            location = " ".join([str(self.round_dec(v)) for v in centroid])
             dimensions = bbox.get_dimensions()
             if self.transformed:
-                dimensions = (dimensions[2], dimensions[1], dimensions[0])
+                try:
+                    self._get_transforms(pcd_path)
+                except CalibrationFileNotFound as exc:
+                    logging.exception("Calibration file not found")
+                    logging.warning("Skipping writing of labels for this point cloud")
+                    return
```

Comment:
I see duplication here. If we called `self._get_transforms` on import, calling it on export in not needed (both matrices were calculated on the first call).

Reply:
Or we should set both matrices to None after each export. Probably it would be better for your version.